lib/oli/lti/platform_external_tools.ex

```diff
@@ -260,4 +272,96 @@ defmodule Oli.Lti.PlatformExternalTools do
   def get_platform_instance(platform_instance_id) do
     Repo.get(PlatformInstance, platform_instance_id)
   end
+
+  @doc """
+  Given a section, return a map where the keys are LTI activity resource IDs and the values are lists of section resources that reference those activities.
+  """
+  def get_section_resources_with_lti_activities(%Section{} = section) do
+    # Step 1: Find all LTI activity registrations
+    lti_activity_registrations =
+      from(ar in ActivityRegistration,
+        join: d in assoc(ar, :lti_external_tool_activity_deployment),
+        select: ar.id
+      )
+      |> Repo.all()
+
+    # Step 2: Get IDs of all section resources in this section that are LTI activities
+    lti_activity_ids =
+      from(sr in SectionResource,
+        join: r in Revision,
+        on: sr.revision_id == r.id,
+        where: sr.section_id == ^section.id and r.activity_type_id in ^lti_activity_registrations,
+        select: r.resource_id
+      )
+      |> Repo.all()
+
+    # Step 3: Find all page section resources that reference these LTI activities
+    page_type_id = ResourceType.id_for_page()
+
+    page_section_resources_with_lti =
+      from(sr in SectionResource,
+        join: r in Revision,
+        on: sr.revision_id == r.id,
+        where:
+          sr.section_id == ^section.id and r.resource_type_id == ^page_type_id and
+            fragment("? && ?", r.activity_refs, ^lti_activity_ids),
+        select: {sr, r}
+      )
+      |> Repo.all()
+
+    # Step 4: Group the results by LTI activity ID
+    page_section_resources_with_lti
+    |> Enum.reduce(%{}, fn {section_resource, revision}, acc ->
+      # Find which LTI activities this page references
+      referenced_lti_activities =
+        Enum.filter(revision.activity_refs, fn ref_id ->
+          ref_id in lti_activity_ids
+        end)
+
+      # Add this section resource to each referenced LTI activity's list
+      Enum.reduce(referenced_lti_activities, acc, fn lti_id, inner_acc ->
+        existing = Map.get(inner_acc, lti_id, [])
+        Map.put(inner_acc, lti_id, [section_resource | existing])
+      end)
+    end)
+  end
+
+  @doc """
+  Given a section, return a map where the keys are LTI activity resource IDs and the values are lists of section resources that reference those activities.
+  """
```

Comment:
I believe this @doc does not describe what this function does.

Reply:
Updated @doc and @spec in bbdc967d9f19396e44c57d7afadfbe3dd5f28c1c